python_modules/libraries/dagster-airbyte/dagster_airbyte/resources.py

```diff
@@ -1171,6 +1173,49 @@ def fetch_airbyte_workspace_data(
             destinations_by_id=destinations_by_id,
         )
 
+    @cached_method
+    def load_asset_specs(
+        self,
+        dagster_airbyte_translator: Optional[DagsterAirbyteTranslator] = None,
+    ) -> Sequence[AssetSpec]:
+        """Returns a list of AssetSpecs representing the Airbyte content in the workspace.
+
+        Args:
+            dagster_airbyte_translator (Optional[DagsterAirbyteTranslator], optional): The translator to use
+                to convert Airbyte content into :py:class:`dagster.AssetSpec`.
+                Defaults to :py:class:`DagsterAirbyteTranslator`.
+
+        Returns:
+            List[AssetSpec]: The set of assets representing the Airbyte content in the workspace.
+
+        Examples:
+            Loading the asset specs for a given Airbyte workspace:
+            .. code-block:: python
+
+                from dagster_airbyte import AirbyteCloudWorkspace
+
+                import dagster as dg
+
+                airbyte_workspace = AirbyteCloudWorkspace(
+                    workspace_id=dg.EnvVar("AIRBYTE_CLOUD_WORKSPACE_ID"),
+                    client_id=dg.EnvVar("AIRBYTE_CLOUD_CLIENT_ID"),
+                    client_secret=dg.EnvVar("AIRBYTE_CLOUD_CLIENT_SECRET"),
+                )
+
+                airbyte_specs = airbyte_workspace.load_asset_specs()
+                defs = dg.Definitions(assets=airbyte_specs, resources={"airbyte": airbyte_workspace}
+        """
+        dagster_airbyte_translator = dagster_airbyte_translator or DagsterAirbyteTranslator()
+
+        return load_airbyte_cloud_asset_specs(
+            workspace=self, dagster_airbyte_translator=dagster_airbyte_translator
+        )
+
+    def sync_and_poll(
+        self, context: Optional[Union[OpExecutionContext, AssetExecutionContext]] = None
```

Comment:
at this point shouldn't it always be AssetExecutionContext?

Reply:
Looking at the implementation, you're right. In it's current state, `sync_and_poll` could be used in an op. 

Is this something we want to support though? cc @yuhan

Reply:
I think we should keep `OpExecutionContext` though - we should raise an error if users are trying to use it in an op. Will add in the implementation in the subsequent PR.

Reply:
I say we don’t support op and throw errors if someone tries to use it in ops. This keeps the surface area small, and we can always loosen it later (it’s better to start restrictive and loosen over time than the opposite).

@maximearmstrong I'm curious the reason why you think we should keep OpExecutionContext. I'm with Chris - I thought at this point, we'd expect the context to be always AssetExecutionContext.

Reply:
@yuhan Yeah, in hindsight I also think removing `OpExecutionContext` is best. My first thought was that all other integrations accept `OpExecutionContext` in their materialization methods, but I think this will only be more confusing for users. Let's just emphasize in the docstring that this method can be used with assets only.

Reply:
Users may not be feeding OpExecutionContext anyways. 

We kept both in other ones likely for backcompat. But we introduced a breaking change in 1.9 [here](https://github.com/dagster-io/dagster/blob/master/CHANGES.md#changes-since-1813-core--02413-libraries) saying
> If you have written helper functions with OpExecutionContext type annotations, they may need to be updated to include AssetExecutionContext depending on your usage.

So I think for net new changes, we can leave out op context.

Besides, that makes me wonder if we should start deprecating all OpExecutionContext from other integrations' materialization methods. This is defn out of scope here, but worth considering.

Reply:
That's good context, thanks!

> Besides, that makes me wonder if we should start deprecating all OpExecutionContext from other integrations' materialization methods. This is defn out of scope here, but worth considering as a followup.

I will make the change for the reworked Fivetran integration, and open a ticket for the other ones.

Reply:
Done in [9636e66](https://github.com/dagster-io/dagster/pull/26432/commits/9636e667d59df49298f19a5e844658303406ae15)